pkg/plugin/packetparser/_cprog/packetparser.c

```diff
@@ -5,39 +5,40 @@
 #include "bpf_helpers.h"
 #include "bpf_endian.h"
 #include "packetparser.h"
+#include "conntrack.c"
+#include "conntrack.h"
 #include "retina_filter.c"
 #include "dynamic.h"
 
 char __license[] SEC("license") = "Dual MIT/GPL";
 
-
-
 struct tcpmetadata {
 	__u32 seq; // TCP sequence number
 	__u32 ack_num; // TCP ack number
+	__u32 tsval; // TCP timestamp value
+	__u32 tsecr; // TCP timestamp echo reply
 	// TCP flags.
 	__u16 syn;
 	__u16 ack;
 	__u16 fin;
 	__u16 rst;
 	__u16 psh;
 	__u16 urg;
-	__u32 tsval; // TCP timestamp value
-	__u32 tsecr; // TCP timestamp echo reply
 };
 
 struct packet
 {
-	// 5 tuple.
+	__u64 ts; // timestamp in nanoseconds
+	__u64 bytes; // packet size in bytes
```

Comment:
this can be __u32, which comes from the length field of `sk_buff` (and is an overkill anyway - packets cannot be this big, even jumbo frames: https://stackoverflow.com/questions/2613734/maximum-packet-size-for-a-tcp-connection)

Reply:
(edited above with extra context)

Reply:
resolved without making the change - bad rebase/merge?

Reply:
oops 🥲

Reply:
oh wait no, i moved this struct to conntrack.c, and it has the right unit

Reply:
![image](https://github.com/user-attachments/assets/7902fc02-e758-4f87-8ab7-9b58650e04b1)